mlir/tools/mlir-miopen-lib/mlir-miopen-lib.h

```diff
@@ -0,0 +1,85 @@
+//===-------------- mlir-miopen-lib.h - C API ---------------------*- C -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM
+// Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef MLIR_C_MLIRMIOPEN_LIBMLIRMIOPEN_H
+#define MLIR_C_MLIRMIOPEN_LIBMLIRMIOPEN_H
+
+#include <stddef.h>
+
+#define MLIRMIOPEN_VERSION_FLAT 0
+
+enum MlirmiopenStatus {
+  MlirmiopenSuccess = 0,
+  MlirmiopenInvalidParam = -1,
+  MlirmiopenInvalidModule = -2,
+  MlirmiopenBuildFailure = -3
```

Comment:
[Recommendation] Use positive values. Use UPPERCASE.
```suggestion
  MLIRMIOPEN_STATUS_SUCCESS = 0,
  MLIRMIOPEN_STATUS_INVALID_PARAM,
  MLIRMIOPEN_STATUS_INVALID_MODULE,
  MLIRMIOPEN_STATUS_BUILD_FAILURE
```

Reply:
Or you have to foollow llvm coding style and it says something different?

Reply:
It does make sense to increment the enum. What do you think @sjw36 ?

Reply:
Ok, it does look like typical C-APIs (see errno) are positive values for errors. I will also make them #defines, which is more typical of C-APIs.

Reply:
Nevermind, I will keep the enum since it is more obvious what the return is.